File: applicatie/src/main/java/com/vdzon/administratie/declaratie/DeclaratieResource.java
package com.vdzon.administratie.declaratie;

import com.vdzon.administratie.util.JsonUtil;
import spark.Spark;

import javax.inject.Inject;


public class DeclaratieResource {

    @Inject
    public DeclaratieResource(DeclaratieService declaratieService) {
        Spark.put("/rest/declaratie/", declaratieService::putDeclaratie, JsonUtil.json());
        Spark.post("/rest/declaratie/", declaratieService::putDeclaratie, JsonUtil.json());
        Spark.delete("/rest/declaratie/:uuid", declaratieService::removeDeclaratie, JsonUtil.json());
    }


}

File: applicatie/src/main/java/com/vdzon/administratie/rest/declaratie/DeclaratieResource.kt
package com.vdzon.administratie.rest.declaratie

import com.vdzon.administratie.util.JsonUtil
import spark.Route
import spark.Spark

import javax.inject.Inject


class DeclaratieResource
@Inject
constructor(declaratieService: DeclaratieService) {
    init {
        Spark.put("/rest/declaratie/", Route { req, res -> declaratieService.putDeclaratie(req, res) }, JsonUtil.json())
        Spark.post("/rest/declaratie/", Route { req, res -> declaratieService.putDeclaratie(req, res) }, JsonUtil.json())
        Spark.delete("/rest/declaratie/:uuid", Route { req, res -> declaratieService.removeDeclaratie(req, res) }, JsonUtil.json())
    }


}
